Cargo.toml

```diff
@@ -3,3 +3,8 @@ name = "oscoin-graph-api"
 version = "0.1.0"
 authors = ["Alexis Sellier <alexis@monadic.xyz>"]
 edition = "2018"
+
+[dependencies]
+
+quickcheck = "*"
```

Comment:
Can this be a `dev-dependencies` perhaps?

Reply:
I am not a Cargo-toml-master, but my understanding is that you can do that if you are relying on `quickcheck` only for test code (or at least this maps from my intuition coming from the Haskell world). Here, however, we are defining a bunch of `Arbitrary` instances for library code, which I would have thought it warrants the import of `quickcheck` as a normal dependency. Or are you saying that in principle one can do something like this, in library code?

```rust

#[cfg(test)]
impl Arbitrary for Foo {
...
}
```

Reply:
That's what I mean :relaxed:

Reply:
Ah, cool, didn't know Rust supported that, will give it a shot!

Reply:
I've never tried it tbh, but it *should* work :stuck_out_tongue_winking_eye:

Reply:
Oh, it does, I have tried already. Interestingly enough what didn't work was your suggested approach (😭) and apparently I wasn't alone in my struggles:

* https://users.rust-lang.org/t/implementing-quickcheck-arbitrary-only-when-testing-crates/12851

* https://stackoverflow.com/questions/36390808/how-to-create-a-quickcheck-arbitrary-of-a-struct-containing-a-reference

Luckily the solution suggested in the first link (which incidentally is the one I have used) worked like a charm 😉

Reply:
Oh strange - I don't quite understand why `#[cfg(test)]` doesn't work, but your solution is :+1: